src/Shared/SpanHelper.cs

```diff
@@ -32,12 +32,13 @@ internal static class SpanHelper
     /// <returns>Resolved span <see cref="Status"/> for the Http status code.</returns>
     public static ActivityStatusCode ResolveSpanStatusForHttpStatusCode(ActivityKind kind, int httpStatusCode)
     {
-        var upperBound = kind == ActivityKind.Client ? 399 : 499;
-        if (httpStatusCode >= 100 && httpStatusCode <= upperBound)
+        var lowerBound = kind == ActivityKind.Client ? 400 : 500;
```

Comment:
The spec also says:

> or 3xx codes with max redirects exceeded), in which case status MUST be set to Error.

Do we need to check for that?

Reply:
Good catch. We might not be able to do this. In instrumentation, there is no way to detect if the response we are looking is the result of final redirect or not. I will follow up on this to see if there is some workaround we could use.

Reply:
Could you create an issue for this?